extensions/opencv/CaptureRTSPFrame.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NIFI_MINIFI_CPP_CAPTURERTSPFRAME_H
+#define NIFI_MINIFI_CPP_CAPTURERTSPFRAME_H
+
+#include <atomic>
+
+#include <core/Resource.h>
+#include <core/Processor.h>
+#include <opencv2/opencv.hpp>
+
+#include <iomanip>
+#include <ctime>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+class CaptureRTSPFrame : public core::Processor {
+
+ public:
+
+  explicit CaptureRTSPFrame(const std::string &name, utils::Identifier uuid = utils::Identifier())
+      : Processor(name, uuid),
+        logger_(logging::LoggerFactory<CaptureRTSPFrame>::getLogger()) {
+  }
+
+  static core::Property RTSPUsername;
+  static core::Property RTSPPassword;
+  static core::Property RTSPHostname;
+  static core::Property RTSPURI;
+  static core::Property RTSPPort;
+  static core::Property ImageEncoding;
+
+  static core::Relationship Success;
+  static core::Relationship Failure;
+
+  void initialize() override;
+  void onSchedule(core::ProcessContext *context, core::ProcessSessionFactory *sessionFactory) override;
```

Comment:
There is no need to call the onTrigger pointer function. That's only there to maintain backwards compatibility. Please use the shared ptr functions only. Thanks